test/js/weak-test.js

```diff
@@ -21,11 +21,16 @@ test('weak', function (t) {
     count++;
   });
 
-  // run weak callback, should dispose
-  gc();
+  var timeout = setTimeout(function () {
```

Comment:
Probably. I copied it from the weak2-test, which tests the internal field variety of weak callbacks. I'll try to change that to `setImmediate` as well.